snooty/semanticparser.py

```diff
@@ -256,7 +257,7 @@ def find_toctree_nodes(
                 toctree_node: Dict[str, SerializableType] = {}
                 if "url" in entry:
                     toctree_node = {
-                        "title": entry.get("title", None),
+                        "title": entry.get("title"),
```

Comment:
This section of code relates to toctree node titles, not the site title, so it is unrelated to your work. This change shouldn't be included in the PR!